enterprise/backend/test/metabase_enterprise/serialization/v2/extract_test.clj

```diff
@@ -1566,3 +1566,24 @@
                (->> (t2/select-one Field :id fk-id)
                     (serdes/extract-one "Field" {})
                     :fk_target_field_id)))))))
+
+(deftest escape-report-test
+  (mt/with-empty-h2-app-db
+    (ts/with-temp-dpc [Collection    {coll1-id :id} {:name "Some Collection"}
+                       Collection    {coll2-id :id} {:name "Other Collection"}
+                       Dashboard     {dash-id :id}  {:name "A Dashboard" :collection_id coll1-id}
+                       Card          {card1-id :id} {:name "Some Card"}
+                       DashboardCard _              {:card_id card1-id :dashboard_id dash-id}
+                       Card          _              {:name "Dependent Card" :collection_id coll2-id
+                                                     :dataset_query {:query {:source-table (str "card__" card1-id)
+                                                                             :aggregation [[:count]]}}}]
+      (testing "Complain about card to available for exporting"
+        (is (= 1
+               (count
+                (mt/with-log-messages-for-level ['metabase-enterprise :warn]
+                  (extract/extract {:targets [["Collection" coll1-id]] :no-settings true :no-data-model true}))))))
```

Comment:
Checking the count is brittle if more log messages are added in the future. I think this should be tighter and check the actual logged string. We have a lot of tests already that check inside the data structure returned by `mt/with-log-messages-for-level` for the exact error message, so I think it's reasonable to do so. 

e.g.
https://github.com/metabase/metabase/blob/bf771f5fb434c810e63b400839c63b3f57e5d506/test/metabase/server/middleware/session_test.clj#L88-L93

Reply:
Fair, yet depending on the message format is preferable to having this break when new logs are added. You have to depend on something

Reply:
How does the new version look? :)

Reply:
Very nice thanks